### ISSUE TITLE
Bug: no-restricted-imports patterns interpret path patterns relative to each file rather than to .eslintrc

### ISSUE THREAD

**Issue by peey** [eslint/eslint#15055]
### Environment

Environment Info:

Node version: v12.18.2
npm version: v7.18.1
Local ESLint version: v7.32.0 (Currently used)
Global ESLint version: Not found
Operating System: linux 5.11.6-051106-generic

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>
<pre>

{
  "root": true,
  "ignorePatterns": [
    "projects/**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "project": [
          "tsconfig.json",
        ],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "no-restricted-imports": ["error", {
          "patterns": ["../*"]
        }]
      }
    }
  ]
}
</pre>
</details>
<details>
<summary>Directory structure</summary>
<pre>
src/dir-a
src/dir-b
.eslintrc
</pre>
</details>
<details>
<summary>Positive example file (eslint should allow)</summary>
<pre>
// src/dir-a/a.ts
import B from "../dir-b/b"
</pre>
</details>
<details>
<summary>Negative example file (eslint should disallow)</summary>
<pre>
// src/dir-a/a.ts
import X from "../../../something"
</pre>
</details>

### What did you expect to happen?

I expected patterns to be relative to location of `.eslintrc`, so that positive example would be allowed and negative one disallowed.

Instead, both are disallowed.

### What actually happened?

Patterns seem to be relative only to location of the file being linted, which are weird semantics. 

This perhaps doesn't matter in the common case where import restrictions are being placed on modules from `node_modules`. But it matters whenever the rule is being made for a path.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @peey, thanks for the issue!

This rule doesn't resolve paths. In your positive example file, it basically works like this:

```js
const ignore = require("ignore"); // v4.0.6

const matcher = ignore().add(configuredPatterns); // ["../*"]

if (matcher.ignores(importSource)) { // true for "../dir-b/b"
    // report error
}
```

`ignore` also doesn't resolve paths and compares `..` literally, as if it was just a directory name (this behavior was changed in ignore v5, it throws an error if a relative path was passed in).

I believe this rule wasn't originally intended to support relative paths, but since it does allow them in configuration we should either document the behavior or consider disallowing them in configuration in a major version.

**Comment by mdjermanovic** [comment]
The [import/no-restricted-paths](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-restricted-paths.md) rule seems better suited for restricting imports within the project.

The core `no-restricted-imports` was designed to target bare specifiers (importing from `node_modules`, or importing built-in modules).

**Comment by peey** [comment]
I'll check if `no-restricted-paths` is suitable for blocking parent-directory imports and get back to you.

**Comment by peey** [comment]
It's actually unclear to me where to put the configuration, could you help me with that? I tried putting `zones` where I'd put `patterns`, i.e.

```
        "no-restricted-imports": [
          "error",
          {
            "zones": [ {
            "target": "../**/*",
            "from": "./**/*"
          }]
          }
        ]
```

but that's likely incorrect because it results in schema validation errors:

```
An unhandled exception occurred: .eslintrc.json#overrides[0]:
        Configuration for rule "no-restricted-imports" is invalid:
        Value {"zones":[{"target":"../**/*","from":"./**/*"}]} should be string.
        Value {"zones":[{"target":"../**/*","from":"./**/*"}]} should NOT have additional properties.
        Value {"zones":[{"target":"../**/*","from":"./**/*"}]} should match some schema in anyOf.
        Value {"zones":[{"target":"../**/*","from":"./**/*"}]} should NOT have additional properties.
        Value [{"zones":[{"target":"../**/*","from":"./**/*"}]}] should match some schema in anyOf.
```

I tried scouring the docs for more information but was unsuccesful

**Comment by mdjermanovic** [comment]
You'll need to install `eslint-plugin-import`:

```bash
npm install eslint-plugin-import --save-dev
```

Then, [configure plugins](https://eslint.org/docs/user-guide/configuring/plugins#configuring-plugins):

```json
{
    "plugins": ["import"]
}
```

and the `import/no-restricted-paths` rule:

```json
{
    "plugins": ["import"],
    "rules": {
        "import/no-restricted-paths": ["error", { ... }]
    }
}
```

**Comment by peey** [comment]
Thanks for helping me configure this. I was able to try out the `no-restricted-paths` rule but it just didn't raise an error on parent directory imports as I was expecting. 

This seems to be a bit of an edge-case. In any case, thanks for clarifying what *no-restricted-imports* is supposed to do. I believe your suggestion of documenting the behaviour is apt and the issue may be closed afterwards. 

I'll continue any discussion relating to excluding parent paths on https://github.com/import-js/eslint-plugin-import/issues repo instead since I guess it's out-of-scope for this repo.

**Comment by nzakas** [comment]
Closing per last comment.
